From 10059617148bbf87b88b794db3e88fb70e90f5d5 Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Tue, 9 Sep 2025 12:19:18 -0400
Subject: [PATCH] state-routes.lua: Make sure the device is still valid after
 doing enum_params()

Since the 'device/store-or-restore-routes' hook is now async, the device might
have been removed before enum_params() finishes. This adds a check to ignore the
hook if the device is not valid.

Fixes #844
---
 src/scripts/device/state-routes.lua | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/scripts/device/state-routes.lua b/src/scripts/device/state-routes.lua
index 0470493c..3349b24a 100644
--- a/src/scripts/device/state-routes.lua
+++ b/src/scripts/device/state-routes.lua
@@ -182,8 +182,15 @@ store_or_restore_routes_hook = AsyncEventHook {
         local selected_routes = {}
         local push_select_routes = false
 
+        -- Make sure the device is still valid
+        if (device:get_active_features() & Feature.Proxy.BOUND) == 0 then
+          transition:advance ()
+          return
+        end
+
         local dev_info = devinfo:get_device_info (device)
         if not dev_info then
+          transition:advance ()
           return
         end
 
-- 
GitLab

